scanpipe/pipes/d2d.py

```diff
@@ -524,3 +576,97 @@ def match_purldb(project, extensions, logger=None):
             start_time=start_time,
         )
         _match_purldb_resource(project, to_resource)
+
+
+def map_javascript(project, logger=None):
+    """Map a packed or minified JavaScript, TypeScript, CSS and SCSS to its source."""
+    project_files = project.codebaseresources.files().only("path")
+
+    from_resources = project_files.from_codebase().filter(
+        extension__in=PROSPECTIVE_JAVASCRIPT_MAP.keys()
+    )
```

Comment:
Suggestion to improve readability:
```
javascript_extensions = PROSPECTIVE_JAVASCRIPT_MAP.keys()
from_resources = project_files.from_codebase().filter(extension__in=javascript_extensions)
```

Reply:
@pombredanne we're only interested in collecting files with these ['.scss', '.js', '.jsx', '.ts', '.d.ts'] extension in `from` side.

Reply:
and the .css may be?

Reply:
@keshav-space my earlier point was about using `.keys()` when the base iteration  on a mapping is always on keys, so there is no need to call `.keys()`

Reply:
> and the .css may be?

To my understanding it's not normal to generate a `.css.map` form `.css` file.
In a usual scenario `.css.map` file will generally come from`.scss`, `.sass` but never directly through `.css `

Reply:
> @keshav-space my earlier point was about using `.keys()` when the base iteration on a mapping is always on keys, so there is no need to call `.keys()`

Won't that be too clever from the point of readability? ;)

Reply:
that's a common idiom, but I am fine either way.